src/main.rs

```diff
@@ -1346,78 +1487,120 @@ fn main() {
                 e.unwrap();
             }
         }
-        let fd = evts[0].data();
-        if fd == listen_sock.as_raw_fd() as u64 {
-            let res = listen_sock.accept();
-            if res.is_err() {
-                eprintln!(
-                    "Failed to accept a connection, error: {:?}",
-                    res.unwrap_err()
+        for evt in evts {
```

Comment:
not sure about the point of this change, only makes it harder to review

Reply:
Should just be more efficient than looping again and going through epoll for no reason.

Reply:
But `evts` always has 1 element? If you want to grab multiple, you need to put multiple event slots there.

Reply:
Oh, I missed that...

Reply:
Fixed ^^

Reply:
You also need to get the return value from epoll.wait, and if it is `Ok(i)`, you process only the first `i` elements of the evts array. (the rest are left as-is and are invalid)

Reply:
I didn't think it mattered since the unwritten events would not match any of the branches, but yeah, makes sense.